packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -0,0 +1,829 @@
+  }
+
+  /// Converts the given [PlatformMenuBarItem] into a data structure accepted by
+  /// the 'flutter/menu' method channel method 'Menu.SetMenu'.
+  ///
+  /// This API is supplied so that implementers of [PlatformMenuBarItem] can share
+  /// this implementation.
+  static Map<String, Object?> serialize(PlatformMenuBarItem item, DefaultPlatformMenuDelegate delegate) {
+    int modifiers = 0;
+    int? logicalKeyId;
+    final ShortcutActivator? shortcut = item.shortcut;
+    if (item.shortcut != null && shortcut is SingleActivator) {
+      if (shortcut.shift) {
+        modifiers |= _kFlutterShortcutModifierShift;
+      }
+      if (shortcut.alt) {
+        modifiers |= _kFlutterShortcutModifierAlt;
+      }
+      if (shortcut.meta) {
+        modifiers |= _kFlutterShortcutModifierMeta;
+      }
+      if (shortcut.control) {
+        modifiers |= _kFlutterShortcutModifierControl;
+      }
+      logicalKeyId = shortcut.trigger.keyId;
+    }
+    return <String, Object?>{
+      _kIdKey: delegate.getId(item),
+      _kLabelKey: item.label,
+      _kEnabledKey: item.onSelected != null,
+      if (shortcut != null) _kShortcutTriggerKey: logicalKeyId,
+      if (shortcut != null) _kShortcutModifiersKey: modifiers,
+    };
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('label', label));
+    properties.add(DiagnosticsProperty<ShortcutActivator?>('shortcut', shortcut, defaultValue: null));
+    properties.add(FlagProperty('enabled', value: onSelected != null, ifFalse: 'DISABLED'));
+  }
+}
+
+/// A widget that represents a menu item that is preconfigured for the given
+/// platform.
+///
+/// This is used to add things like the "About" and "Quit" menu items to a
+/// platform menu.
+///
+/// The [type] enum determines which type of platform defined menu will be
+/// added.
+///
+/// This is most useful on a macOS platform where there are many different types
+/// of platform provided menu items in the standard menu setup.
+///
+/// In order to know if a [PlatformProvidedMenuItem] is available on a
+/// particular platform, call [PlatformProvidedMenuItem.hasMenu].
+///
+/// If the platform does not support the given [type], then the menu item will
+/// throw an [ArgumentError].
+///
+/// See also:
+///
+///  * [PlatformMenuBar] which takes these items for inclusion in a
+///    platform-rendered menu bar.
+class PlatformProvidedMenuItem extends PlatformMenuBarItem with DefaultPlatformMenuDelegateSerializer {
+  /// Creates a const [PlatformProvidedMenuItem] of the appropriate type. Throws if the
+  /// platform doesn't support the given default menu type.
+  ///
+  /// The [type] argument is required.
+  const PlatformProvidedMenuItem({
+    required this.type,
+    this.enabled = true,
+  }) : super(
+          label: '', // The label is ignored for standard menus.
```

Comment:
This leads me to the question: How are these menu entries localized? Does MacOS just provide a localized version? So If my app only supports German (and my app-provided menus are all in German), if I run this on a Chinese MacOS, will my own items appear in german, but the MacOS provided ones will be in Chinese? Is there a way to configure that? Should we document that?

Reply:
Yes, macOS provides localized versions.  As far as I know there's no way to configure it other than in the XCode files for the app.  We probably should document it, but I'd have to understand it first, and I don't yet.